compiler/luci-micro/luci-interpreter/src/core/CMakeLists.txt

```diff
@@ -1,24 +1,31 @@
 set(SOURCES
     "${LUCI_INTERPRETER_INCLUDE_DIR}/luci_interpreter/core/DataType.h"
     "${LUCI_INTERPRETER_INCLUDE_DIR}/luci_interpreter/core/Tensor.h"
-    "${LUCI_INTERPRETER_INCLUDE_DIR}/luci_interpreter/core/CircleMicroReader.h"
     Kernel.h
     KernelParams.h
     RuntimeGraph.h
     RuntimeGraph.cpp
     RuntimeModule.h
-    Tensor.cpp
-    CircleMicroReader.cpp)
+    Tensor.cpp)
+
+set(READER_SOURCE
+        "${LUCI_INTERPRETER_INCLUDE_DIR}/luci_interpreter/core/CircleMicroReader.h"
+        "${LUCI_INTERPRETER_INCLUDE_DIR}/luci_interpreter/core/CircleMicroReaderHelper.h"
+        CircleMicroReader.cpp
+        CircleMicroReaderHelper.cpp
+        )
+
+add_library(circle_reader STATIC ${READER_SOURCE})
+target_link_libraries(circle_reader PUBLIC circle_schema)
+target_include_directories(circle_reader PUBLIC "${GENERATED_INCLUDE_DIR}")
+target_include_directories(circle_reader PUBLIC "${LUCI_INTERPRETER_INCLUDE_DIR}")
 
 add_library(${LUCI_INTERPRETER_CORE} STATIC ${SOURCES})
 if (NOT NNCC_LIBRARY_NO_PIC)
     set_target_properties(${LUCI_INTERPRETER_CORE} PROPERTIES POSITION_INDEPENDENT_CODE ON)
 endif(NOT NNCC_LIBRARY_NO_PIC)
```

Comment:
by the way!
Note, we could remove this option.
it was needed when we used common sources of the interpreter.